### PR TITLE
Added echoback

### PR DESCRIPTION
**Pull request by sp-bechara**

### CODE REVIEW

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c [hunk 1]
```diff
@@ -114,9 +102,18 @@ int main(void)
     /* USER CODE END WHILE */
 
     /* USER CODE BEGIN 3 */
-	  HAL_UART_Receive_IT(&huart2, &receivedData, 1);
-	  HAL_UART_Transmit(&huart2, msg1, sizeof(msg1),HAL_MAX_DELAY);
-	  HAL_Delay(1000);
+//	  if (HAL_UART_Receive(&huart2, (uint8_t*)&receivedChar, 1, HAL_MAX_DELAY) == HAL_OK) {
```
Comment (kaushaleinfo) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: Remove comment

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c [hunk 2]
```diff
@@ -114,9 +102,18 @@ int main(void)
...
+//	        // Echo back the received character
+//	        HAL_UART_Transmit(&huart2, (uint8_t*)&receivedChar, 1, HAL_MAX_DELAY);
+//	      }
+
+	  HAL_UART_Receive_IT(&huart2, (uint8_t*)&receivedChar, 1);
```
Comment (kaushaleinfo) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: Place this IT function before while(1)

File: Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c [hunk 3]
```diff
@@ -209,8 +208,7 @@ void USART2_IRQHandler(void)
   /* USER CODE END USART2_IRQn 0 */
   HAL_UART_IRQHandler(&huart2);
   /* USER CODE BEGIN USART2_IRQn 1 */
-
-	  inter();
+  inter();
```
Comment (BhuvaHiren) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c]: Do this call require argument?

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c [hunk 4]
```diff
@@ -238,39 +235,8 @@ static void MX_GPIO_Init(void)
...
-			HAL_UART_Transmit(&huart2, msg9, sizeof(msg9),HAL_MAX_DELAY);
-		}
-	if(receivedData=='8'){
-			HAL_UART_Transmit(&huart2, msg10, sizeof(msg10),HAL_MAX_DELAY);
-		}
-	if(receivedData=='9'){
-			HAL_UART_Transmit(&huart2, msg11, sizeof(msg11),HAL_MAX_DELAY);
-		}
-	receivedData=0;
+void inter(int *receivedChar){
```
Comment (kaushaleinfo) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: Change function name

File: Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c (hunk 3 above)
Comment (kaushaleinfo) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c]: Put this function at receive interrupt code

**Review by BhuvaHiren — CHANGES_REQUESTED** [review]

**Review by kaushaleinfo — CHANGES_REQUESTED** [review]

File: Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c (hunk 3 above)
Comment (sp-bechara) [reply Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c]: done

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c (hunk 1 above)
Comment (sp-bechara) [reply Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: done

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c (hunk 2 above)
Comment (sp-bechara) [reply Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: done

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c (hunk 4 above)
Comment (sp-bechara) [reply Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: done

File: Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c (hunk 3 above)
Comment (sp-bechara) [reply Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c]: done

File: Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c [hunk 5]
```diff
@@ -208,7 +208,7 @@ void USART2_IRQHandler(void)
   /* USER CODE END USART2_IRQn 0 */
   HAL_UART_IRQHandler(&huart2);
   /* USER CODE BEGIN USART2_IRQn 1 */
-  inter();
+  //ISR();
```
Comment (kaushaleinfo) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/stm32f4xx_it.c]: Why did you remove this function call and where did you put this function call ?

File: Task/MCU/Floor1_USART_inputdata/Core/Src/main.c [hunk 6]
```diff
@@ -91,10 +92,10 @@ int main(void)
   MX_GPIO_Init();
   MX_USART2_UART_Init();
   /* USER CODE BEGIN 2 */
-
+#if ECHOBACK
```
Comment (kaushaleinfo) [inline Task/MCU/Floor1_USART_inputdata/Core/Src/main.c]: Use #ifdef macro instead of #if

**Review by BhuvaHiren — APPROVED** [review]
